src/components/RestockDialog.vue

```diff
@@ -0,0 +1,320 @@
+                    @click="restockAndClose()"
+                    variant="tonal"
+                >
+                    restock and close
+                </v-btn>
+            </v-card-actions>
+        </v-card>
+    </v-dialog>
+</template>
+
+<script lang="ts" setup>
+import { useClient } from '@/graphql/client'
+import { useAppStore } from '@/store/app'
+import { errorMessages } from '@/strings/errorMessages'
+import {
+    pushErrorNotification,
+    pushErrorNotificationIfNecessary,
+} from '@/util/errorHandler'
+import { asyncComputed } from '@vueuse/core'
+import { storeToRefs } from 'pinia'
+import { computed, ref } from 'vue'
+
+/**
+ * Props definition for the component.
+ *
+ * @property productId - The ID of the product.
+ * @property [preselectedProductVariantId] - The ID of the preselected product variant.
+ * @property [maximumBatchSize] - The maximum batch size allowed for restocking.
+ */
+const props = defineProps({
+    productId: {
+        type: String,
+        required: true,
+    },
+    preselectedProductVariantId: {
+        type: String,
+    },
+    maximumBatchSize: {
+        type: Number,
+    },
+})
+
+/**
+ * Emits definition for the component.
+ *
+ * Event: close - The dialog should be closed.
+ * Event: restocked - The selected product variant was successfully restocked.
+ */
+const emit = defineEmits<{
+    (event: 'close'): void
+    (event: 'restocked'): void
+}>()
+
+/**
+ * Reactive references and values from the Pinia store.
+ */
+const { activeUserRoleIsEitherAdminOrEmployee } = storeToRefs(useAppStore())
+
+/**
+ * The GraphQL client to use for all GraphQL requests.
+ */
+const client = useClient()
+
+/**
+ * Async computed property to fetch the product from the catalog service.
+ * Note that the returned object essentially only contains the product's variants.
+ */
+const getProductWithItsVariantsOnlyQuery = asyncComputed(
+    async () => {
+        if (activeUserRoleIsEitherAdminOrEmployee.value) {
+            return client.getProductWithItsVariantsOnly({
+                id: props.productId,
+            })
+        }
+
+        return null
+    },
+    null,
+    {
+        onError: (e) =>
+            pushErrorNotification(
+                errorMessages.getProductWithItsVariantsOnly,
+                e
+            ),
+        shallow: false,
+    }
+)
+
+/**
+ * Computed property for extracting the product variants from the fetched product data.
+ */
+const productVariants = computed(() => {
+    if (getProductWithItsVariantsOnlyQuery.value !== null) {
+        return getProductWithItsVariantsOnlyQuery.value.product.variants.nodes
+    }
+
+    return []
+})
+
+/**
+ * Computed property for generating an array of amount options based on the maximum batch size.
+ */
+const amountOptions = computed(() => {
+    const defaultMaximumLength = 100
+    const length =
+        props.maximumBatchSize !== undefined
+            ? props.maximumBatchSize
+            : defaultMaximumLength
```

Comment:
why not use the default field when defining the prop?

Reply:
"Fixed" with [this commit](https://github.com/MiSArch/frontend/pull/37/commits/8cd627d1c7be8ddc783c9b60d64100b59ae5b640)